python-client/giskard/slicing/filters.py

```diff
@@ -0,0 +1,106 @@
+import logging
+from scipy import stats
+
+from giskard.datasets import Dataset
+from giskard.ml_worker.testing.tests.performance import (
+    test_diff_f1,
+    test_diff_rmse,
+    test_diff_accuracy,
+    test_diff_recall,
+    test_diff_precision,
+)
+
+
+class DataSliceFilter:
+    """Selects slices based on their significance."""
+
+    def filter(self, slices):
+        raise NotImplementedError()
+
+
+class InternalTestFilter(DataSliceFilter):
+    def __init__(self, model, dataset, test_name=None, threshold=0.1):
+        self.dataset = dataset
+        self.model = model
+        self.threshold = threshold
+
+        if test_name == "f1":
+            self.test = test_diff_f1
+        if test_name == "accuracy":
```

Comment:
let's use `elif` to skip the subsequent equality checks once the test is initialized